kodilla-hibernate/src/test/java/com/kodilla/hibernate/manytomany/dao/CompanyDaoTestSuite.java

```diff
@@ -0,0 +1,137 @@
+
+    @Autowired
+    EmployeeDao employeeDao;
+
+    @Test
+    public void testSaveManyToMany() {
+        //Given
+        Employee johnSmith = new Employee("John", "Smith");
+        Employee stephanieClarckson = new Employee("Stephanie", "Clarckson");
+        Employee lindaKovalsky = new Employee("Linda", "Kovalsky");
+
+        Company softwareMachine = new Company("Software Machine");
+        Company dataMaesters = new Company("Data Maesters");
+        Company greyMatter = new Company("Gray Matter");
+
+        softwareMachine.getEmployees().add(johnSmith);
+        dataMaesters.getEmployees().add(stephanieClarckson);
+        dataMaesters.getEmployees().add(lindaKovalsky);
+        greyMatter.getEmployees().add(johnSmith);
+        greyMatter.getEmployees().add(lindaKovalsky);
+
+        johnSmith.getCompanies().add(softwareMachine);
+        johnSmith.getCompanies().add(greyMatter);
+        stephanieClarckson.getCompanies().add(dataMaesters);
+        lindaKovalsky.getCompanies().add(dataMaesters);
+        lindaKovalsky.getCompanies().add(greyMatter);
+
+        //When
+        companyDao.save(softwareMachine);
+        int softwareMachineId = softwareMachine.getId();
+        companyDao.save(dataMaesters);
+        int dataMaestersId = dataMaesters.getId();
+        companyDao.save(greyMatter);
+        int greyMatterId = greyMatter.getId();
+
+        //Then
+        Assert.assertNotEquals(0, softwareMachineId);
+        Assert.assertNotEquals(0, dataMaestersId);
+        Assert.assertNotEquals(0, greyMatterId);
+
+        //CleanUp
+        try {
+          companyDao.deleteById(softwareMachineId);
+          companyDao.deleteById(dataMaestersId);
+          companyDao.deleteById(greyMatterId);
+         } catch (Exception e ) {
+              //do nothing
+        }
+    }
+
+    @Test
+    public void testQueryCompanyBy3FirstLetters() {
+        //Given
+        Company softwareMachine = new Company("Software Machine");
+        Company dataMaesters = new Company("Data Maesters");
+        Company greyMatter = new Company("Gray Matter");
+        Company softSkin = new Company("SoftSkin");
+
+        companyDao.save(softwareMachine);
+        companyDao.save(dataMaesters);
+        companyDao.save(greyMatter);
+        companyDao.save(softSkin);
+
+        //When
+        List<Company> companySof = companyDao.findCompanyByFirst3Letters("Sof");
+
+        //Then
+        try {
+            Assert.assertEquals(2, companySof.size());
+        } finally {
+            companyDao.deleteAll();
+        }
+    }
+
+    @Test
+    public void testQueryEmployeeByLastname() {
+        //Given
+        Employee johnSmith = new Employee("John", "Smith");
+        Employee stephanieClarckson = new Employee("Stephanie", "Clarckson");
+        Employee lindaKovalsky = new Employee("Linda", "Kovalsky");
+        Employee lindaBruebeck = new Employee("Martha", "Kovalsky");
+
+        Company softwareMachine = new Company("Software Machine");
```

Comment:
Żeby przetestować to zapytanie, nie potrzebujesz tworzyć żadnych firm i przypisywać ich pracownikom. Wystarczą Ci same encje pracowników...